tests/unit/test_utils.py

```diff
@@ -155,6 +156,16 @@ def test_color(self):
 class TestFetch(object):
     """ Remote reference from fmf github """
 
+    def test_fetch_default_branch(self):
```

Comment:
Could you add a test which changes branch back to default branch?
1. fetch node from explicit, no-default branch
2. fetch node from default branch (without specifying ref:)

Reply:
Also extending `test_out_of_sync_ref` with ref=None will be nice

Reply:
I've added a new test `test_switch_branches` to cover branch switching. Regarding `test_out_of_sync_ref`, here the default branch probably does not make sense, because `__DEFAULT__` is a ref to `origin/master` and thus it cannot be out of sync from it. If the fresh `origin/master` is fetched when cache expires `__DEFAULT__` will be still correct.